### PR TITLE
fix(metric-cards): recursively define card size [MA-1446]

### PR DESCRIPTION
**Pull request by mihai-peteu**
# Summary

## PR Checklist

* [ ] **Naming & Structure:** the files and package structure use the conventions outlined in the [Creating a Package docs](https://github.com/Kong/public-ui-components/blob/main/docs/creating-a-package.md).
* [ ] **Tests pass:** check the output of all package unit and/or component tests.
  * [ ] If this PR is the result of a bug, test coverage was added accordingly.
* [ ] **Functional:** all changes do not break existing APIs, but if so, a BREAKING CHANGE commit is in place to bump the major version.
* [ ] **Conventional Commits** all commits follow the conventional commit standards [outlined in the main README](https://github.com/Kong/public-ui-components#committing-changes).
* [ ] **Docs:** includes a technically accurate README, and the docs have been updated accordingly based on the changes in this PR.

### CODE REVIEW

File: packages/analytics/metric-cards/src/types/metric-card.ts [hunk 1]
```diff
@@ -24,6 +25,6 @@ export interface MetricCardContainerOptions {
   fallbackDisplayText: string
   hasTrendAccess: boolean
   loading: boolean
-  cardSize?: string
```
Comment (filipgutica) [inline packages/analytics/metric-cards/src/types/metric-card.ts:27]: Should individual cards even have cardSize?

**Review by filipgutica — CHANGES_REQUESTED** [review]

File: packages/analytics/metric-cards/src/types/metric-card.ts (hunk 1 above)
Comment (mihai-peteu) [reply packages/analytics/metric-cards/src/types/metric-card.ts:27]: Oh good point, this was copy/pasta from khcp-ui 😬 . Will remove.

Comment (mihai-peteu) [reply packages/analytics/metric-cards/src/types/metric-card.ts:27]: The cardSize prop is drilled down from the parent MetricCardContainer. So I should keep it here as well, no?

Comment (mihai-peteu) [reply packages/analytics/metric-cards/src/types/metric-card.ts:27]: I'll keep cardSize on MetricCardContainerOptions, and drop it on MetricCardDef

**Comment by mihai-peteu** [comment]
@jillztom included the tweak we discussed in Slack:

https://github.com/Kong/public-ui-components/pull/186/commits/052eb7df94414aa6442d7128f6a504d2b8c1f737

**Review by jillztom — APPROVED** [review]
Looks good

**Review by filipgutica — APPROVED** [review]

**Pull request merged**